File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        
        origin = x
        rev_num = 0
        
        while x > 0:
            rev_num = rev_num*10 + x%10
            x = x//10
            
        if rev_num == origin:
            return True
        
        return False
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0){
            return false;
        }
        
        int reverse_x = 0;
        int origin = x;
        
        while (x != 0){
            reverse_x = (reverse_x * 10) + (x % 10);
            x = x / 10;
        }
        
        return reverse_x == origin;
    }
}